### PR TITLE
feature(suggestions): Implement accept and reject functionality

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: scripts/core/editor3/components/suggestions/SuggestionPopup.jsx [hunk 1]
```diff
@@ -33,9 +38,20 @@ export class SuggestionPopup extends Component {
                 this.setState({error: gettext('An error occured, please try again.')});
             });
     }
+
+    onAccept() {
+        // TODO: close widget
```
Comment (petrjasek) [inline scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: not sure if you have to close it, when you accept the entity should be gone and thus the popup should be gone, same on reject..

Comment (ioanpocol) [reply scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: Yes, somehow it is not gone.

File: scripts/core/editor3/reducers/suggestions.jsx [hunk 2]
```diff
@@ -115,6 +119,58 @@ const pasteAddSuggestion = (state, {content, data}) => {
...
+        type = entity.get('type');
+
+        const applySuggestion = type === 'ADD_SUGGESTION' && accepted || type === 'DELETE_SUGGESTION' && !accepted;
+        const newSelection = selection.merge({
+            anchorOffset: start + i + (applySuggestion ? 0 : 1),
+            focusOffset: start + i + (applySuggestion ? 0 : 1),
+            isBackward: false
+        });
+
+        editorState = EditorState.acceptSelection(editorState, newSelection);
```
Comment (petrjasek) [inline scripts/core/editor3/reducers/suggestions.jsx]: maybe you should use `forceSelection` here to fix the popup closing, but not sure

File: scripts/core/editor3/components/suggestions/SuggestionPopup.jsx (hunk 1 above)
Comment (tomaskikutis) [reply scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: FYI this line is responsible for closing it https://github.com/superdesk/superdesk-client-core/blob/master/scripts/core/editor3/components/HighlightsPopup.jsx#L271

File: scripts/core/editor3/reducers/suggestions.jsx (hunk 2 above)
Comment (ioanpocol) [reply scripts/core/editor3/reducers/suggestions.jsx]: After I rebased it is working fine.

File: scripts/core/editor3/components/suggestions/SuggestionPopup.jsx (hunk 1 above)
Comment (ioanpocol) [reply scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: After I rebased it is working fine.

File: scripts/core/editor3/components/suggestions/SuggestionPopup.jsx [hunk 3]
```diff
@@ -56,7 +72,7 @@ export class SuggestionPopup extends Component {
         };
 
         return (
-            <Dropdown key={this.props.keyForDropdown} open={true}>
```
Comment (tomaskikutis) [inline scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: This is required so before [author data is fetched](https://github.com/superdesk/superdesk-client-core/pull/2079/files#diff-4b54af773b8aaa818f599e3086f73861R33) there is no empty box displayed.

File: scripts/core/editor3/components/suggestions/SuggestionPopup.jsx [hunk 4]
```diff
@@ -33,9 +38,20 @@ export class SuggestionPopup extends Component {
...
+
+    onReject() {
+        // TODO: close widget
+        this.props.rejectSuggestion(this.props.suggestion.selection);
+    }
+
     render() {
         if (this.state.error !== null) {
-            return <Dropdown key={this.props.keyForDropdown} open={true}>{this.state.error}</Dropdown>;
+            return <div open={true}>{this.state.error}</div>;
```
Comment (tomaskikutis) [inline scripts/core/editor3/components/suggestions/SuggestionPopup.jsx]: Did you mean to leave `open={true}`?

**Pull request merged**